### ISSUE TITLE
Disguise removal warhead options

### ISSUE THREAD

**Issue by MikillRosen** [Phobos-developers/Phobos#70]
The idea is a warhead INI flag that removes the disguise of spies, allowing forces to kill the spies automatically. With this, you could make, for an example, an unarmed support unit that removes the disguise of enemy spies, making armed units attack them without the need of the player's orders.

**Comment by secsome** [comment]
That can be easily done, try 
```cpp
if (pWHExt->RemoveDisguise) {
			if (pThis->Target->WhatAmI() == AbstractType::Infantry)
			{
				auto pInf = abstract_cast<InfantryClass*>(pThis->Target);
				if (pInf->IsDisguised())
				{
					bool bIsAlliedWith = pThisHouse->IsAlliedWith(pInf);
					if(pWHExt->RemoveDisguise_AffectAllies || (!pWHExt->RemoveDisguise_AffectAllies && !bIsAlliedWith))
						pInf->ClearDisguise();
				}
			}
		}
```
in BulletClass_Detonate

**Comment by secsome** [comment]
TODO : Take CellSpread into account

**Comment by Otamaa** [comment]
Use ALex Helper from Ares , this can help with cellspread stuffs 
//this 
#include <Helpers/Iterators.h>
#include <InfantryClass.h>


#include "../../Utilities/Helpers.Alex.h"

	if (pWHExt->RemoveDisguise) {
		auto coords = *pCoordsDetonation;
		auto CellSpread = pWH->CellSpread;
		if (CellSpread) {
                                                    //this
			const auto items = Helpers::Alex::getCellSpreadItems(coords, CellSpread, true);
			for (auto member : items) {

				if (member->WhatAmI() == AbstractType::Infantry)
				{
					auto pInf = abstract_cast<InfantryClass*>(pThis->Target);

					if (pInf->IsDisguised()) {

						bool bIsAlliedWith = pThisHouse->IsAlliedWith(pInf);
			if (pWHExt->RemoveDisguise_AffectAllies || (!pWHExt->RemoveDisguise_AffectAllies && !bIsAlliedWith)) {

							pInf->ClearDisguise();
						}
					}
				}
			}
		}
	}

**Comment by Metadorius** [comment]
@secsome @Otamaa mind opening up a pull request?

**Comment by Otamaa** [comment]
ah , sorry  not familiar using git stuffs
tho i can help if needed

**Comment by Metadorius** [comment]
> ah , sorry not familiar using git stuffs
> tho i can help if needed

Basically fork and clone the repo, make a new branch from `develop` (could name it something like `feature/disguise-removal-wh`), commit this code, push the branch, then go to GitHub and create a pull request to develop.

**Comment by Metadorius** [comment]
Alternatively @secsome could do it if he wants.

**Comment by secsome** [comment]
@Metadorius  I've introduced the templates and opened a pull request, so we can make further works easier I think.

**Comment by Metadorius** [comment]
Implemented in #73.
